### ISSUE TITLE
WidgetsBinding.instance?.addObserver(this); Null pointer exception

### ISSUE THREAD

**Issue by musti-91** [supabase-community/supabase-flutter#166]
Hello,

It seems that supabase package is not supported by having null pointer operator.

Flow: 
just installed supabase package in new flutter project start failing regarding this error: 

```
/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_required_state.dart:43:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/homebrew/Caskroom/flutter/3.0.4/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
    WidgetsBinding.instance?.addObserver(this);
                   ^
/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_required_state.dart:49:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/homebrew/Caskroom/flutter/3.0.4/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
    WidgetsBinding.instance?.removeObserver(this); 

```

Flutter SDK: 3.0.5
Supabase version: ^0.3.3
Dart version: 2.17.6

It needs to be updated to remove null pointer operators.

**Comment by dshukertjr** [comment]
Thanks for opening this issue @musti-91 !

If you read it carefully, the message you are getting is just a warning, but is not anything that prevents the app from launching. The `?` is left there intentionally to support Flutter 2.X versions, which requires them. 

I would like to close this issue as we have a duplicate one here https://github.com/supabase-community/supabase-flutter/issues/138

**Comment by musti-91** [comment]
I actually it's not. it's an error. I cannot run the app only after using the Supabase.initialize.
the same is happening in the boilerplate 
https://github.com/supabase-community/supabase-flutter

**Comment by dshukertjr** [comment]
@musti-91 
Having an extra `?` is not an error, and that `?` is necessary to support Flutter 2.X versions where `WidgetsBinding.instance` is a type `WidgetsBinding?`. 

I only see warning, but no error within the message you provided. If your app is not launching properly, the cause lies somewhere else. 

It would be great if you could open a new issue with some explanations on how your app fails to launch!

**Comment by musti-91** [comment]
```
Launching lib/main.dart on iPhone 13 in debug mode...
lib/main.dart:1
Xcode build done.                                            7.4s
Failed to build iOS app
Error output from Xcode build:
↳
    2022-08-15 13:00:17.302 xcodebuild[81643:3628055] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
    2022-08-15 13:00:17.302 xcodebuild[81643:3628055] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
    ** BUILD FAILED **
Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/j1/r4jqdj0d1tv9gwy7qgfxz7q80000gn/T/flutter_tools.QZJ93i/flutter_ios_build_temp_dirzKtd06/temporary_xcresult_bundle
/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_required_state.dart:43:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/homebrew/Caskroom/flutter/3.0.4/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
        WidgetsBinding.instance?.addObserver(this);
                       ^
/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_required_state.dart:49:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/homebrew/Caskroom/flutter/3.0.4/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
        WidgetsBinding.instance?.removeObserver(this);
                       ^
/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_state.dart:32:5: Error: The method 'onReceivedAuthDeeplink' isn't defined for the class 'SupabaseAuthState<T>'.
- 'SupabaseAuthState' is from 'package:supabase_flutter/src/supabase_auth_state.dart' ('/opt/homebrew/Caskroom/flutter/3.0.4/flutter/.pub-cache/hosted/pub.dartlang.org/supabase_flutter-0.3.3/lib/src/supabase_auth_state.dart').
package:supabase_flutter/src/supabase_auth_state.dart:1
    Try correcting the name to the name of an existing method, or defining a method named 'onReceivedAuthDeeplink'.
        onReceivedAuthDeeplink(uri);
        ^^^^^^^^^^^^^^^^^^^^^^
    Failed to package /Users/alroomi/Developer/youtube/supabase_auth.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order
    Result bundle written to path:
    	/var/folders/j1/r4jqdj0d1tv9gwy7qgfxz7q80000gn/T/flutter_tools.QZJ93i/flutter_ios_build_temp_dirzKtd06/temporary_xcresult_bundle
Could not build the application for the simulator.
Error launching application on iPhone 13.
Exited
```
I don't mind opening another issue but if you could help here would be great to resolve the issue. the full logs mentioned above.
It's indeed not the issue of optional binding But it seems something to do with 
`onReceivedAuthDeeplink(uri);`

Hope that can helps!!!

**Comment by musti-91** [comment]
commenting this line helps launching the app.
```32  // onReceivedAuthDeeplink(uri); // supabase_auth_state.dart```

**Comment by bdlukaa** [comment]
can you try it on the latest version?
